### ISSUE TITLE
AM2301 Issue

### ISSUE THREAD

**Issue by G4KCM** [hydromisc/hydromisc#8]
It seems that the AM2301 doesn't work on either input it just returns 0.0 for both temperature and humidity.
I'm wondering if this is also related to the version of the compiler?  Can't think why looking at the code but I must admit I can't get my head around it.  I suspect it might be that the timing is adrift as I see on inspection the device is an AM2301A not AM2301. 

I wonder if you could let me have a functioning binary to test and/or advise which version of the ESP-idf was used in ther meantime I will see if I can get an origional AM2301/DHT21 somewhere.

Thanks

**Comment by hydromisc** [comment]
I built with `ESP-IDF v4.1-dirty`, though I'm surprised that would matter. My binary wouldn't work for you, since the Wi-Fi credentials are compiled in. Note that you may need an external pull-up resistor (from the red wire to yellow) if the wires are long. If it still doesn't work, then can you scope the yellow wire?

**Comment by G4KCM** [comment]
> I built with `ESP-IDF v4.1-dirty`, though I'm surprised that would matter. My binary wouldn't work for you, since the Wi-Fi credentials are compiled in. Note that you may need an external pull-up resistor (from the red wire to yellow) if the wires are long. If it still doesn't work, then can you scope the yellow wire?

Hi, appreciate you getting back to me…  I don’t think the ESP-IDF version will matter either.  First thing I tried was a pull-up and that made no difference also with a scope on the data line the sensor is sending back data but it returns 0.0 for both temperature and humidity.  I think the reason is that the ‘new’ AM2301 that I have purchased is actually labelled as an AM2301A the insides of the housing are not the same as the ‘original’ AM2301 as it looks like they have got something like and AHT10 inside and and some firmware mimic the old device and translate to the single comms line.  In fact the AM2301A and an AM2302 I have are identical inside so I suspect that’s the issue.   I have just ordered some ‘original’ AM2301 from AliExpress 🤞

Just one question, I see there is a dht.c library that supports the AM230x devices included within the esp-idf. Is there a specific reason that you didn’t use that?  I may have a play using that library to see if it will read the AM2301A devices whilst I wait for the slow boat from China.  Thanks again for a great bit of work.
